activerecord/lib/active_record/tasks/database_tasks.rb

```diff
@@ -169,7 +169,7 @@ def for_each(databases)
         return if database_configs.count == 1
 
         database_configs.each do |db_config|
-          yield db_config.name
+          yield db_config.name, db_config
```

Comment:
This method returns just the db_config name because it's not a real db_config. It's db_config without a fully generated configuration_hash. We can't use it to dump the schema cache in the tasks. We must establish a connection and get the real db_config after the task is initialized the way the other tasks do.